Sources/MapboxDirections/RouteResponse.swift

```diff
@@ -58,12 +58,28 @@ extension RouteResponse: Codable {
             routes = try decoder.decode([Route].self, from: matchesData)
         }
         
-        var waypoints: [Waypoint]?
+        var waypoints: [Route.Waypoint]?
         
         if let tracepoints = response.tracepoints {
             let filtered = tracepoints.compactMap { $0 }
             let tracepointsData = try encoder.encode(filtered)
-            waypoints = try decoder.decode([Waypoint].self, from: tracepointsData)
+            waypoints = try decoder.decode([Route.Waypoint].self, from: tracepointsData)
+            
+            // Push tracepoints down to individual legs to reflect waypoint_index.
+            if let routes = routes {
+                let sorted = zip(filtered, waypoints!).sorted {
+                    $0.0.waypointIndex < $1.0.waypointIndex
+                }
+                for (index, route) in routes.enumerated() {
+                    let legSeparators = sorted.filter {
+                        return $0.0.matchingIndex == index
+                    }.map { $0.1 }
+                    
+                    assert(legSeparators.count == route.legs.count + 1)
```

Comment:
I assume this is something we can expect to be guaranteed?

Reply:
There will be fewer leg separators if any of the waypoints in the request had `separatesLegs` equal to `false`.